src/screens/TopicPageScreen/elements/BottomSheetFollow.tsx

```diff
@@ -40,18 +41,22 @@ const BottomSheetFollow = forwardRef((props: BottomSheetFollowProps, ref: Ref<RB
     onClose
   } = props;
   const {followTopic} = useChatClientHook();
+  const {updateChannelList, deleteChannelList} = useRootChannelListHook();
   const {updateFollowTopicDiscoveryContext} = useDiscovery();
 
   const handleFollowTopic = async ({type: followTypeParam}: {type: Follow}) => {
     const isIncognito = followTypeParam === 'incognito';
 
     const followTypeCheck = (targetType: Follow) => followType === targetType || followType === '';
 
+    const channelType = followTypeParam === 'incognito' ? 'ANON_TOPIC' : 'TOPIC';
+
     const follow = async (type: Follow) => {
       setFollowType(type);
       setIsFollow(true);
       try {
         await followTopic(topicName, isIncognito);
+        updateChannelList(topicId, channelType);
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `updateChannelList` and `deleteChannelList` methods are being used here. However, there is no error handling around these calls. If either of these operations fail, the user will not be notified and the application state could become inconsistent. Consider adding error handling around these operations.
```diff
    const handleFollowTopic = async ({type: followTypeParam}: {type: Follow}) => {
      const isIncognito = followTypeParam === 'incognito';

      const followTypeCheck = (targetType: Follow) => followType === targetType || followType === '';

     const channelType = followTypeParam === 'incognito' ? 'ANON_TOPIC' : 'TOPIC';

      const follow = async (type: Follow) => {
        setFollowType(type);
        setIsFollow(true);
        try {
          await followTopic(topicName, isIncognito);
+         try {
            updateChannelList(topicId, channelType);
+         } catch (error) {
+           console.error('Failed to update channel list:', error);
+         }
          updateFollowTopicDiscoveryContext(true, {topic_id: topicId}, true);
        } catch (error) {
          if (__DEV__) {
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->